robbie:

  # Publish all joint states -----------------------------------
  joint_state_controller:
   type: joint_state_controller/JointStateController
   publish_rate: 50  

  whole_body_controller:
   type: effort_controllers/JointTrajectoryController
   joints:
    - stab_joint
    - knee_joint
    - hip_joint
    - lhm_torso_joint
    - shoulder_left_joint
    - shoulder_right_joint
    - elbow_left_joint
    - elbow_right_joint
   gains:
    shoulder_left_joint: {p: 100, i: 0.1, d: 10.0}
    shoulder_right_joint: {p: 100, i: 0.1, d: 10.0}
    elbow_left_joint: {p: 100, i: 0.1, d: 10.0}
    elbow_right_joint: {p: 100, i: 0.1, d: 10.0}
    hip_joint: {p: 33000, i: 6.75, d: 1.688}
    lhm_torso_joint: {p: 33000, i: 6.75, d: 1.688}
    stab_joint: {p: 36000, i: 6.5, d: 1.625}
    knee_joint: {p: 36000, i: 6.5, d: 1.625}

   state_publish_rate: 50
   action_monitor_rate: 30
   stop_trajectory_duration: 0

    # lhm_torso_joint: {p: 4000, i: 0.1, d: 50.0}


  stand_controller:
   type: effort_controllers/JointTrajectoryController
   joints:
    - shoulder_left_joint
    - shoulder_right_joint
    - hip_joint
    - lhm_torso_joint
    - stab_joint
    - knee_joint
   gains:
    shoulder_left_joint: {p: 100, i: 0.1, d: 10.0}
    shoulder_right_joint: {p: 100, i: 0.1, d: 10.0}
    hip_joint: {p: 33000, i: 6.75, d: 1.688}
    lhm_torso_joint: {p: 4000, i: 0.1, d: 50.0}
    stab_joint: {p: 36000, i: 6.5, d: 1.625}
    knee_joint: {p: 36000, i: 6.5, d: 1.625}


    constraints:
      stopped_velocity_tolerance: 0.05
   state_publish_rate: 50
   action_monitor_rate: 30
   stop_trajectory_duration: 0

  stand_controller_tuning:
   type: effort_controllers/JointTrajectoryController
   joints:
    - shoulder_left_joint
    - shoulder_right_joint
    - hip_joint
    - lhm_torso_joint
    - stab_joint
    - knee_joint
   gains:
    # shoulder_left_joint: {p: 100, i: 0.1, d: 10.0}
    # shoulder_right_joint: {p: 100, i: 0.1, d: 10.0}
    # hip_joint: {p: 33000, i: 0.215, d: 0.054}
    # lhm_torso_joint: {p: 4000, i: 0.1, d: 50.0}
    # stab_joint: {p: 2000, i: 0.1, d: 10.0}
    # knee_joint: {p: 36000, i: 6.5, d: 1.625}

    shoulder_left_joint: {p: 0, i: 0, d: 0}
    shoulder_right_joint: {p: 0, i: 0, d: 0}
    hip_joint: {p: 0, i: 0, d: 0}
    lhm_torso_joint: {p: 0, i: 0, d: 0}
    stab_joint: {p: 0, i: 0, d: 0}
    knee_joint: {p: 0, i: 0, d: 0}

    constraints:
      stopped_velocity_tolerance: 0.05
   state_publish_rate: 50
   action_monitor_rate: 30
   stop_trajectory_duration: 0



   # shankwheel_effort_controller:
   #   type: diff_drive
   #   left_wheel: 'wheel_left_joint'
   #   right_wheel: 'wheel_right_joint'




  # leftWheel_effort_controller:
  #  type: effort_controllers/JointEffortController
  #  joint: wheel_left_joint
  #  pid: {p: 100.0, i: 0.1, d: 10.0}

  # rightWheel_effort_controller:
  #  type: effort_controllers/JointEffortController
  #  joint: wheel_right_joint
  #  pid: {p: 100.0, i: 0.1, d: 10.0}


  # lhm_diff_controller:
  #   type: "diff_drive_controller/DiffDriveController"
  #   left_wheel: 'lhm_wheel_left_joint'
  #   right_wheel: 'lhm_wheel_right_joint'
  #   pose_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
  #   twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
    # base_frame_id: shank_footprint
    # wheel_radius: 0.102175
    # wheel_seperation: 0.425

  lhm_right_wheel_effort_controller:
   type: effort_controllers/JointEffortController
   joint: lhm_wheel_right_joint
   pid: {p: 100.0, i: 0.1, d: 10.0}
  lhm_left_wheel_effort_controller:
   type: effort_controllers/JointEffortController
   joint: lhm_wheel_left_joint
   pid: {p: 100.0, i: 0.1, d: 10.0}


    # shoulder_left_joint: {p: 100, i: 0.1, d: 10.0}
    # shoulder_right_joint: {p: 100, i: 0.1, d: 10.0}
    # hip_joint: {p: 2000, i: 0.1, d: 10.0}
    # lhm_torso_joint: {p: 4000, i: 0.1, d: 50.0}
    # stab_joint: {p: 2000, i: 0.1, d: 10.0}
    # knee_joint: {p: 2000, i: 0.1, d: 10.0}
